tt-media-server/tests/server_tests/base_test.py

```diff
@@ -0,0 +1,117 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+
+from abc import abstractmethod
+import os
+import asyncio
+import time
+import traceback
+from typing import Any, Dict
+
+from tests.server_tests.test_classes import TestConfig
+
+
+class BaseTest:
+    def __init__(self, config: TestConfig, targets: Dict[str, Any]):
+        self.config = config
+        self.targets = targets
+        self.service_port = os.getenv("SERVICE_PORT", "8000")
+        self.timeout = config.get("timeout")
+        self.retry_attempts = config.get("retry_attempts")
+        self.break_on_failure = config.get("break_on_failure")
+        self.logs = []
+        self.retry_delay = config.get("retry_delay")
+        
+    def run_tests(self):
+        last_exception = None
+        
+        for attempt in range(self.retry_attempts + 1):
+            try:
+                print(f"Running tests (attempt {attempt + 1}/{self.retry_attempts + 1})")
```

Comment:
I think it would be good practice to use logger instead of print methods:
```suggestion
                import logging
                // finish importing other stuff
                
                logger = logging.getLogger(__name__)
                
                class BaseTest(ABC):
                // code
                logger.info(f"Running tests (attempt {attempt + 1}/{self.retry_attempts + 1})")
```